src/main/java/io/aiven/kafka/connect/common/output/parquet/ParquetOutputWriter.java

```diff
@@ -0,0 +1,115 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.connect.common.output.parquet;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.Collection;
+import java.util.Map;
+import java.util.Objects;
+
+import org.apache.kafka.connect.sink.SinkRecord;
+
+import io.aiven.kafka.connect.common.config.OutputField;
+import io.aiven.kafka.connect.common.output.OutputStreamWriter;
+import io.aiven.kafka.connect.common.output.OutputWriter;
+
+import io.confluent.connect.avro.AvroData;
+import io.confluent.connect.avro.AvroDataConfig;
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.parquet.avro.AvroParquetWriter;
+import org.apache.parquet.hadoop.ParquetFileWriter;
+import org.apache.parquet.hadoop.ParquetWriter;
+import org.apache.parquet.io.OutputFile;
+import org.apache.parquet.io.PositionOutputStream;
+
+public class ParquetOutputWriter extends OutputWriter {
+
+    static final int PARQUET_PAGE_SIZE = 64 * 1024;
+
+    private ParquetWriter<GenericRecord> parquetWriter;
+
+    private Schema parquetSchema;
+
+    private final SinkRecordConverter sinkRecordConverter;
+
+    private final ParquetSchemaBuilder parquetSchemaBuilder;
+
+    public ParquetOutputWriter(final Collection<OutputField> fields,
+                               final OutputStream out,
+                               final Map<String, String> externalConfig) {
+        super(new ParquetPositionOutputStream(out), new OutputStreamWriterStub(), externalConfig);
+        final var avroData = new AvroData(new AvroDataConfig(externalConfig));
+        this.sinkRecordConverter = new SinkRecordConverter(fields, avroData);
+        this.parquetSchemaBuilder = new ParquetSchemaBuilder(fields, avroData);
+    }
+
+    @Override
+    public void writeRecord(final SinkRecord record) throws IOException {
+        if (Objects.isNull(parquetWriter)) {
+            final var parquetConfig = new ParquetConfig(externalConfiguration);
+            parquetSchema = parquetSchemaBuilder.buildSchema(record);
+            parquetWriter = AvroParquetWriter.<GenericRecord>builder(new ParquetOutputFile())
+                    .withSchema(parquetSchema)
+                    .withWriteMode(ParquetFileWriter.Mode.OVERWRITE)
+                    .withDictionaryEncoding(true)
+                    .withConf(parquetConfig.parquetConfiguration())
+                    .withCompressionCodec(parquetConfig.compressionCodecName())
+                    .withPageSize(PARQUET_PAGE_SIZE)
+                    .build();
+        }
```

Comment:
Is it needed to initialize this lazily instead of initialization in the constructor?